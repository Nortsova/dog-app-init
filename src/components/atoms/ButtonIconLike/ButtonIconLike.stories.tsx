import type { Meta, StoryObj } from '@storybook/react';
import { ButttonIconLike } from '.';

const meta = {
  title: 'Atoms/ButttonIconLike',
  component: ButttonIconLike,
  argTypes: {
    state: {
      options: ['default'],
      control: 'radio'
    }
  }
} satisfies Meta<typeof ButttonIconLike>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    state: 'default'
  }
};
