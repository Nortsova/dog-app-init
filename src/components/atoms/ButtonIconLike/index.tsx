import { SvgIcon, styled } from '@mui/material';
import theme from '../../../theme/theme';

import IconLike from '../../../assets/icons/IconLike.svg?react';

const BtnLike = styled('button')({
  padding: theme.spacing(1, 1.4),
  borderRadius: 30,
  backgroundColor: theme.palette.primary.main,
  border: theme.palette.primary.main,
  boxShadow: `2px 2px 0px 0px ${theme.palette.text.secondary}`,
  '&:hover': {
    backgroundColor: theme.palette.action.hover,
    border: theme.palette.action.hover
  },
  '&:active': {
    backgroundColor: theme.palette.action.selected,
    border: theme.palette.action.selected
  }
});

export const ButttonIconLike = () => (
  <BtnLike>
    <SvgIcon
      component={IconLike}
      sx={{ width: 16, height: 17 }}
      inheritViewBox
    />
  </BtnLike>
);
