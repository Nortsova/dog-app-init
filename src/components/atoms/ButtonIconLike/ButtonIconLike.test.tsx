import { composeStories } from '@storybook/react';
import * as stories from './ButtonIconLike.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('ButttonIconLike', () => {
  const { Primary } = composeStories(stories);
  testSnapshot(<Primary />);
});
