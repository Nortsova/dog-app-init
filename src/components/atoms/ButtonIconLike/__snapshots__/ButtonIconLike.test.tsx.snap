// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButttonIconLike should match snapshot Component 1`] = `
.emotion-0 {
  padding: 8px 11.2px;
  border-radius: 30px;
  background-color: #921FED;
  border: #921FED;
  box-shadow: 2px 2px 0px 0px #000000;
}

.emotion-0:hover {
  background-color: #A239F4;
  border: #A239F4;
}

.emotion-0:active {
  background-color: #B568F2;
  border: #B568F2;
}

<div>
  <button
    class="emotion-0"
  >
    <test-file-stub
      aria-hidden="true"
      classname="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-dfwo49-MuiSvgIcon-root"
      focusable="false"
    />
  </button>
</div>
`;
